Added Wson 14 1EP

suzizecat opened the pull request:
Added the WSON-14_1EP used in KiCad/kicad-library#1300
Based on [TI datasheet](http://www.ti.com/lit/ml/mpds421/mpds421.pdf) specified as WSON in [LP3947](http://www.ti.com/lit/ds/symlink/lp3947.pdf)

![image](https://user-images.githubusercontent.com/1909563/26980730-3b2c9eea-4d33-11e7-9f3d-f8546d1e4e24.png)

![image](https://user-images.githubusercontent.com/1909563/26980758-58cbb5a8-4d33-11e7-90bb-0edcf76c943e.png)

3D file not set yet because @hackscribble is working on it

suzizecat committed:
Added WSON-14-1EP from TI

Commit before fix

suzizecat committed:
Fixed KLC issues

ghost commented:
Hi @suzizecat I've compared the model I've made with the footprint, to check that we interpreted the datasheet the same way.  

Sizes match, but I think the footprint needs to be reoriented ...

- This image in the datasheet is the view from underneath the package looking up at the pins and EP. So looking down onto the footprint and model, pin 1 would be the bottom left pin in your image above, not top left where the pin 1 silkscreen and fab markers are.

![image](https://user-images.githubusercontent.com/7076027/26981519-59b9d874-4d2d-11e7-8d78-8d09a3ae371e.png)

- For consistency with the other footprints in the repo, it should also be rotated 90 degrees clockwise, so that pin 1 is at top left rather than bottom left.

![image](https://user-images.githubusercontent.com/7076027/26981732-f3cb1bee-4d2d-11e7-9921-721ac4bb1dea.png)

suzizecat commented:
Well, i may have misunderstood that datasheet, i'm not very familiar with mechanical views... I thought that it was strange to got the first pin in this configuration but, why not ?

So, let's rotate that !

![image](https://user-images.githubusercontent.com/1909563/26982125-a8a94d8e-4d37-11e7-9272-f6a805680c9c.png)

EDIT : 
I placed the ref des on the top of the component

suzizecat committed:
Fixing DS misunderstood

90° Rotation, pin reassignment and labels placements

ghost commented:
:smile: it fits (ignore the missing pin on the model - I was testing something).  When this PR is merged, I will finalise the model and include it in my new PR.

![image](https://user-images.githubusercontent.com/7076027/26982345-0ee104b4-4d30-11e7-9dc1-89a584941160.png)

suzizecat commented:
Yay ! 😺 
Could you tell me the exact name of your 3D model to allow me to link it to my footprint ?
EDIT : NVM if you gonna link it yourself (if that's what you mean by including it)

poeschlr commented:
The 3d model should have the same name as the footprint. (Unless there is a good reason to do it differently.)
So the 3dpath should look like: ${KISYS3DMOD}/Housings_SON.3dshapes/WSON-14_1EP_4.0x4.0mm_Pitch0.5mm.wrl

suzizecat commented:
Well, i don't know if there is a WSON-14 without EP, but then it should have the same name, no ? So the path shouldn't be ${KISYS3DMOD}/Housings_SON.3dshapes/WSON-14_4.0x4.0mm_Pitch0.5mm.wrl ?

poeschlr commented:
As far as i know, the 3d model includes the exposed pad so it is definetly the ep version. If there is a wson 14 without ep than the 3d model should have the exposed pad removed.
@hackscribble do you model the exposed pad?

suzizecat commented:
Well, KiCAD models goes way too far in perfection for me 😄 
I'll update that tomorrow or monday then.

Thanks to you for the review and thanks to @hackscribble for his work !

ghost commented:
The EP is in the 3D model.

poeschlr commented:
@suzizecat if you add the 3d model information to this footprint it might be ready to merge.

suzizecat commented:
I added the 3D model as it follow
EDIT : i added the prefix
![image](https://user-images.githubusercontent.com/1909563/27025281-3f3a579c-4f5a-11e7-9969-47899c47cc5e.png)

suzizecat committed:
Added 3D model information

poeschlr commented:
I seem to have missed some things.
The documentation field should contain more information than just the datasheet.

The fab layer has coordinates that look like y=-1.973276 ("top" side) should this be on y=-2
The vertical lines are not straight. (x direction of the top corners looks similar to the y coordinate.)

Courtyard should have a clearance of 0.25 all around
Rectangle from (-2.45, -2.25) to (2.45, 2.25) (Your x direction is ok. y direction currently -2.22 and 2.28.)

suzizecat committed:
Cleaning dimensions

suzizecat committed:
Fixed courtyard

Set at 0.25mm relatively to component and pads

suzizecat commented:
I fixed the weird dimensions (probably made by the `--fix` option) and  i properly set the courtyard to 0.25mm relatively to any part of the component (either as it is in the fab layer, and the pads). If the courtyard should be set regarding only the pads or the real component size, please let me know.

Thanks for your reviews =)

suzizecat committed:
Added documentation field

poeschlr commented:
Looks very nice now.
To answer your question: Yes the courtyard clearance is either measured with respect to the pads or to the component outline (fab layer) depending on what results in the larger courtyard area.

Thanks for your great contribution and thanks for fixing all these pesky little issues i threw at you.

suzizecat commented:
Thanks to you to allow KiCad to be such a great tool by looking for these sneaky issues ;-) !
